File: Java/SQLite/app/src/main/java/com/ob/sqlite/MainActivity.java
package com.ob.sqlite;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.ob.sqlite.databinding.ActivityMainBinding;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity
{
    ActivityMainBinding binding;
    DBHelper dbHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        Button btnAdd = binding.btnAdd;
        Button btnFetch = binding.btnFetch;
        Button btnUpdate = binding.btnUpdate;
        Button btnDelete = binding.btnDelete;

        btnAdd.setOnClickListener(view1 -> {
            Person person1 = new Person(1,"Onur","Bilke");
            Person person2 = new Person(2,"Cansu","Bilke");
            Person person3 = new Person(3,"Ada","Bilke");
            Person person4 = new Person(4,"Deniz","Bilke");

            dbHelper.addPerson(person1);
            dbHelper.addPerson(person2);
            dbHelper.addPerson(person3);
            dbHelper.addPerson(person4);
        });

        btnFetch.setOnClickListener(view1 -> {
            ArrayList<Person> persons = dbHelper.getPersons();

            for (Person person : persons) {
                System.out.println("Id : " + person.id + " Name : " + person.name +" Surname : " + person.surname);
            }
        });

        btnUpdate.setOnClickListener(view1 -> {
            Person person = new Person(1,"On","Bi");
            dbHelper.updatePerson(person);
        });

        btnDelete.setOnClickListener(view1 -> dbHelper.deletePerson(1));

        dbHelper = new DBHelper(this);
    }
}

File: Kotlin/SQLite/app/src/main/java/com/ob/sqlite/MainActivity.kt
package com.ob.sqlite

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.ob.sqlite.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity()
{
    private lateinit var binding: ActivityMainBinding
    private lateinit var dbHelper: DBHelper

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        dbHelper = DBHelper(this)
    }

    fun onAdd(view:View) {
        val person1 = Person(1,"Onur","Bilke")
        val person2 = Person(2,"Cansu","Bilke")
        val person3 = Person(3,"Ada","Bilke")
        val person4 = Person(4,"Deniz","Bilke")

        dbHelper.addPerson(person1)
        dbHelper.addPerson(person2)
        dbHelper.addPerson(person3)
        dbHelper.addPerson(person4)
    }

    fun onFetch(view:View) {
        val persons = dbHelper.getPersons()

        for (person in persons) {
            println("Id : ${person.id} Name : ${person.name} Surname : ${person.surname}")
        }
    }

    fun onUpdate(view:View) {
        val person = Person(1,"On","Bi")
        dbHelper.updatePerson(person)
    }

    fun onDelete(view:View) {
        dbHelper.deletePerson(1)
    }
}